### ISSUE TITLE
Two questions about the ouput

### ISSUE THREAD

**Issue by DaVilla7** [MIC-DKFZ/nnUNet#930]
Hi, thank you for the nnUnet model.

I run a binary segmentation (0:' bg', 1: 'fg') task and I have two questions,

1. I notice that the output contains two channels, which I think should mean the background and the foreground, but as far as I know, it is normal to just output one channel of logits and get a binary prediction by sigmoid, I wonder what's the purpose to set two channels for a binary segmentation. The log shows the 'Average global foreground Dice:', is it means it just calculates the dice score within the channel of foreground?

2. From the log of a 2D generated net, it seems the output has five scals, i.e.,
    (0): Conv2d(480, 2, kernel_size=(1, 1), stride=(1, 1), bias=False)
    (1): Conv2d(480, 2, kernel_size=(1, 1), stride=(1, 1), bias=False)
    (2): Conv2d(256, 2, kernel_size=(1, 1), stride=(1, 1), bias=False)
    (3): Conv2d(128, 2, kernel_size=(1, 1), stride=(1, 1), bias=False)
    (4): Conv2d(64, 2, kernel_size=(1, 1), stride=(1, 1), bias=False)
    (5): Conv2d(32, 2, kernel_size=(1, 1), stride=(1, 1), bias=False)
 is it means it calculates a multi-scale loss or dice with the output? If so, how does it works?

**Comment by FabianIsensee** [comment]
Hey there,
about 1: it doesn't matter. Both work. In the current implementation nnU-Net works for n flasses (incl background) and the corner case of just one class isn't covered.
about 2: Please take a look at the paper regarding deep supervision (online methods). I am just one guy and there are so many people asking questions that are answered in the paper. 
Best,
Fabian

**Comment by DaVilla7** [comment]
@FabianIsensee 
Thank for your answer, it helps a lot.
About the deep supervision, I think 'no new U-net' should have follow the classical U-net architecture while deep supervision is not included.
Anyway I understand, thank you for your reply.

**Comment by DaVilla7** [comment]
@FabianIsensee 
Sorry to bother you for a second time,

(0): Conv2d(480, 2, kernel_size=(1, 1), stride=(1, 1), bias=False)
(1): Conv2d(480, 2, kernel_size=(1, 1), stride=(1, 1), bias=False)
(2): Conv2d(256, 2, kernel_size=(1, 1), stride=(1, 1), bias=False)
(3): Conv2d(128, 2, kernel_size=(1, 1), stride=(1, 1), bias=False)
(4): Conv2d(64, 2, kernel_size=(1, 1), stride=(1, 1), bias=False)
(5): Conv2d(32, 2, kernel_size=(1, 1), stride=(1, 1), bias=False)

For the six output, they are different scales, how do they be fused and generate a single output? If upsampling and concat them, the total channel will be 2*6 = 12 but I did not see a followed 1*1 Conv layer to reduce the channel to 2.
As far as I know 'deep supervision'  fuses the output by upsampling + 1*1 Conv to keep the channel size equal to the channel of input img.

So in this case how does it work?

**Comment by FabianIsensee** [comment]
> About the deep supervision, I think 'no new U-net' should have follow the classical U-net architecture while deep supervision is not included.

The experimental results disagree with this opinion ;-)

> For the six output, they are different scales, how do they be fused and generate a single output?

As the paper says, each of these outputs is compared to a matching downsampled version of the segmentation and the losses are added together

**Comment by Karol-G** [comment]
Closing this issue as it seems solved. Feel free to open the issue again, if you have further questions.
